pkg/bundle/publish/manifests.go

```diff
@@ -0,0 +1,309 @@
+package publish
+
+import (
+	"fmt"
+	"hash/fnv"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	operatorv1alpha1 "github.com/openshift/api/operator/v1alpha1"
+	"github.com/openshift/oc/pkg/cli/image/imagesource"
+	"gopkg.in/yaml.v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// Copied from https://github.com/openshift/oc/blob/5d8dfa1c2e8e7469d69d76f21e0a166a0de8663b/pkg/cli/admin/catalog/mirror.go#L549
+// Changes made are breaking ICSP and Catalog Source generation into difference functions
```

Comment:
Any reason why you're doing this? If the image digests are provided, why not use the default ICSPs and CatalogSources generated by `oc` libs?

Reply:
@estroz we are creating ICSPs for user defined images and releases so I broke it up so we will only generate catalog sources for catalog index images.